core/app/models/spree/store_credit.rb

```diff
@@ -43,7 +43,9 @@ class Spree::StoreCredit < Spree::PaymentSource
   before_destroy :validate_no_amount_used
   validate :validate_no_amount_used, if: :discarded?
 
-  attr_accessor :action, :action_amount, :action_originator, :action_authorization_code, :update_reason
+  belongs_to :store_credit_reason, class_name: 'Spree::StoreCreditReason', inverse_of: :store_credits
+
+  attr_accessor :action, :action_amount, :action_originator, :action_authorization_code, :store_credit_reason
```

Comment:
I don't think `store_credit_reason` here is needed anymore

Reply:
done